compliant/src/lib.rs

```diff
@@ -0,0 +1,1171 @@
+    let url = "https://raw.githubusercontent.com/huggingface/kernel-builder/refs/heads/main/build-variants.json";
+    let response = reqwest::get(url)
+        .await
+        .context("Failed to connect to variants endpoint")?;
+
+    if !response.status().is_success() {
+        return Err(CompliantError::VariantsFetchError(format!(
+            "HTTP error: {}",
+            response.status()
+        ))
+        .into());
+    }
+
+    let variants_config: VariantsConfig = response
+        .json()
+        .await
+        .context("Failed to parse variants JSON")?;
+
+    let mut cuda_variants = Vec::new();
+    cuda_variants.extend(variants_config.x86_64_linux.cuda);
+    cuda_variants.extend(variants_config.aarch64_linux.cuda);
+
+    #[cfg(feature = "enable_rocm")]
+    let rocm_variants = variants_config.x86_64_linux.rocm;
+
+    #[cfg(not(feature = "enable_rocm"))]
+    let rocm_variants = Vec::new();
+
+    Ok((cuda_variants, rocm_variants))
+}
+
+/// Synchronous wrapper for fetching variants. This avoids spreading async/await throughout
+/// the codebase. Used for compatibility with sync contexts.
+fn fetch_variants_sync() -> Result<(Vec<String>, Vec<String>)> {
+    let rt = tokio::runtime::Runtime::new().context("Failed to create Tokio runtime")?;
+    rt.block_on(fetch_compliant_variants())
+}
+
+/// Cached variants to avoid repeatedly fetching the same data
+pub static COMPLIANT_VARIANTS: Lazy<(Vec<String>, Vec<String>)> = Lazy::new(|| {
+    match fetch_variants_sync() {
+        Ok(variants) => variants,
+        Err(e) => {
+            // We still need to handle initialization errors, but without process::exit
+            // This still panics but at least gives proper error context
+            panic!("Failed to fetch compliant variants: {}", e);
+        }
+    }
+});
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Variant {
+    torch_version: String,
+    cxx_abi: String,
+    compute_framework: String,
+    arch: String,
+    os: String,
+}
+
+impl fmt::Display for Variant {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{}-{}-{}-{}-{}",
+            self.torch_version, self.cxx_abi, self.compute_framework, self.arch, self.os
+        )
+    }
+}
+
+impl Variant {
+    pub fn from_name(name: &str) -> Option<Self> {
+        let parts: Vec<&str> = name.split('-').collect();
+        if parts.len() < 5 {
+            return None;
+        }
+        // Format: torch{major}{minor}-{cxxabi}-{compute_framework}-{arch}-{os}
+        Some(Variant {
+            torch_version: parts[0].to_string(),
+            cxx_abi: parts[1].to_string(),
+            compute_framework: parts[2].to_string(),
+            arch: parts[3].to_string(),
+            os: parts[4].to_string(),
+        })
+    }
+}
+
+/// Struct to hold repository list result
+#[derive(Serialize)]
+pub struct RepoListResult {
+    pub repositories: Vec<String>,
+    pub count: usize,
```

Comment:
How does this differ from `repositories.len()`?

Reply:
removed in latest commits